dandiapi/api/models/upload.py

```diff
@@ -52,7 +52,7 @@ class Meta:
     size = models.PositiveBigIntegerField()
 
     @abstractclassmethod
-    def object_key(cls, upload_id, dandiset: Dandiset):  # noqa: N805
+    def object_key(cls, upload_id, dandiset: Dandiset | None = None):  # noqa: N805
```

Comment:
```
    @abstractstaticmethod
    def object_key(upload_id, dandiset: Dandiset | None = None):  # noqa: N805
```
Is that a thing?

Reply:
Apparently yes https://docs.python.org/3/library/abc.html#abc.abstractstaticmethod

But you're really supposed to do this:

```python
    @staticmethod
    @abstractmethod
    def object_key(upload_id, dandiset: Dandiset | None = None):
```

I'll fix it.

Reply:
> New in version 3.2.
> 
> Deprecated since version 3.3

Poor decorator though it was going to make it big in Pythonworld 😭